### ISSUE TITLE
Make tag definitions closer to class definitions

### ISSUE THREAD

**Issue by judofyr** [somebee/imba#54]
The proposal here is to make tag definitions with a *capital* name behave more like classes. For instance:

```imba
tag Note
  def render
    <self> …

# is just sugar for
class Note < Element
  def render
    <self> …
```

This would then be local to the current file (or you could annotate it with `export` or `global` as with regular classes). The tag usage would then be a plain constructor:

```imba
<Note title="Hello" />
# compiles to:
(new Note).setTitle("Hello").end()
```

Lowercase tag names would still be global (as they are today) and should be used sparingly.

Note that this is very similar to how React works: https://babeljs.io/repl/#?experimental=true&evaluate=false&loose=false&spec=true&code=%3CDiv%20%2F%3E%3B%0A%3Cdiv%20%2F%3E%3B.

**Comment by somebee** [comment]
How do you deal with collisions? One of the things I like about tags is that I can have a tag named todo *and* a class named Todo, which allows for natural separation of views / models without names like TodoView and TodoModel. Could it behind the scenes compile tag Note to class NoteTag or something? But this would be trickier with importing etc

**Comment by judofyr** [comment]
I don't mind calling my components for `TodoView`.

**Comment by somebee** [comment]
I see. You can currently define `tag todo` without needing to worry about it overwriting or interacting with any `var todo` that might be in the same scope/file. Would it not be confusing/inconsistent to allow this for uppercase tags? Anyhow, the suggestion itself is definately good.

If you have a tag defined at Util.List - how do you spawn it? `<Util:List>`? Using `<Util.List>` will not work due to our flag syntax.

**Comment by somebee** [comment]
This is now implemented in the current 0.15.0 alpha.
